pkg/eestream/decode.go

```diff
@@ -86,12 +86,10 @@ func (dr *decodedReader) Read(p []byte) (n int, err error) {
 		dr.currentStripe++
 	}
 
-	// copy what data we have to the output
+	// copy what data we have to the output, and shrink the remaining buffer
 	n = copy(p, dr.outbuf)
-	// slide the remaining bytes to the beginning
-	copy(dr.outbuf, dr.outbuf[n:])
-	// shrink the remaining buffer
-	dr.outbuf = dr.outbuf[:len(dr.outbuf)-n]
+	dr.outbuf = dr.outbuf[n:]
```

Comment:
i think this is a memory leak. isn't it? over time, the allocated size of the backing buffer will grow and grow, even though the slice that is referenced is smaller. perhaps append will occasionally lop off old allocations but that will increase gc pressure.

Reply:
okay not a memory leak. looks like infectious.Decode never appends to the existing outbuf but just reuses it. 

this does increase garbage collector pressure though. by sliding the outbuf out to the right the existing code will allocate a new buffer when it no longer has enough space. i'd be surprised if this actually makes things more performant

Reply:
oh, looks like you talked it through with egon. i didn't see the earlier convo. it is indeed a matter of avoiding new buffer allocations.

Reply:
> this does increase garbage collector pressure though. by sliding the outbuf out to the right the existing code will allocate a new buffer when it no longer has enough space

That's right if you append and you want to reuse the buffer, however I didn't find where that append happens with my quick view, I thought that it was only read once.

btw, I found this reading code and trying to understand some logic, not looking exactly to improve performance or anything else, I thought that it was a matter of something that escaped from a code review.